nanoFramework.DependencyInjection/Microsoft/Extensions/DependencyInjection/ServiceCollectionServiceExtensions.cs

```diff
@@ -80,6 +80,30 @@ public static IServiceCollection AddSingleton(this IServiceCollection services,
             return services;
         }
 
+        /// <summary>
+        /// Adds a singleton service of the type specified in <paramref name="serviceType"/>
+        /// with the specified <paramref name="factory"/> to the
+        /// specified <see cref="IServiceCollection"/>.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection"/> to add the service to.</param>
+        /// <param name="serviceType">The type of the service to register.</param>
+        /// <param name="factory">A factory used for creating service instances.</param>
+        /// <returns>A reference to this instance after the operation has completed.</returns>
+        /// <seealso cref="ServiceLifetime.Singleton"/>
+        /// <exception cref="ArgumentNullException"><paramref name="services"/> can't be <see langword="null"/>.</exception>
+        public static IServiceCollection AddSingleton(this IServiceCollection services, Type serviceType, ImplementationFactoryDelegate factory)
```

Comment:
Stupid question: is this method aligned with the .NET one?

Reply:
Good point! I updated methods & comments.